content/en/blog/2021/wasm-api-alpha/index.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Announcing the alpha availability of WebAssembly Plugins
+description: Announcement of the alpha availability of Wasm plugins in Istio.
+publishdate: 2021-11-25
+attribution: "Daniel Grimm(Red Hat), Pengyuan Bian(Google), Takeshi Yoneda(Tetrate)"
+
+keywords: [wasm,extensibility,WebAssembly]
+---
+
+[At the Istio 1.9 release](https://istio.io/latest/blog/2021/wasm-progress/), we introduced experimental support for Wasm module distribution and wasm extensions ecosystem repository for canonical examples and use cases of Wasm extension development. Over the past 9 months, the Istio, Envoy, and Proxy-Wasm communities have continued our joint efforts to make WebAssembly (Wasm) extensibility stable, reliable, and easy to adopt, and we are pleased to announce Alpha support for Wasm extensibility at Istio 1.12 release! Let’s walk through the updates to Wasm support through the Istio 1.12 release.
+
+## New WasmPlugin API
+
+With the new WasmPlugin API in the extensions.istio.io namespace, we’re introducing a new high-level API for extending the functionality of the Istio Proxy with custom WebAssembly (Wasm) modules. This effort builds on the excellent work that has gone into the [Proxy-Wasm](https://github.com/proxy-wasm) specification and implementation over the last two years. From now on, you no longer need to use EnvoyFilter resources to add custom Wasm modules to your proxies. Instead, you can now use WasmPlugin directly:
+
+{{< text yaml >}}
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: your-filter
+spec:
+  selector:
+    matchLabels:
+      app: server
+  phase: AUTHN
+  priority: 10
+  pluginConfig:
+    someSetting: true
+    someOtherSetting: false
+    youNameIt:
+    - first
+    - second
+  url: docker.io/your-org/your-filter:1.0.0
+{{< /text >}}
+
+There are a lot of similarities and a few differences between WasmPlugin and EnvoyFilter, so let’s go through the fields one by one. The above example deploys a Wasm module to all workloads (including gateway pods) that match it’s `selector` - this very much works the same as in an EnvoyFilter.
+
+The next field below that is the `phase`. This determines where in the proxy’s filter chain the Wasm module will be injected. We have defined four distinct phases for injection:
+
+AUTHN: this is prior to any istio authentication filters, meaning it is executed even before the remote mTLS certificates are verified.
+AUTHZ: just after the istio authentication filters and before any authorization filters, i.e. before AuthorizationPolicies have been applied.
+STATS: 
```

Comment:
@dgn What should I put on this line?

Reply:
https://github.com/istio/istio.io/pull/10530/commits/082c727c28946aca754ede4287d1c1c287a564f9 done